documentation/docs/payid.mdx

```diff
@@ -0,0 +1,74 @@
+---
+id: payid
+title: PayID
+---
+
+import { Mermaid } from './Mermaid'
+
+By default, Open Payments interactions centre around an [account](./accounts)
+resource and all APIs are relative to the account URL.
+
+The account URL is [discovered](./discovery) through a
+[Payment Pointer](./terminology#payment-pointer) which is an account identifer
+for Interledger-enabled accounts.
+
+Said differently, any account identified by a Payment Pointer can be paid into
+using the [STREAM](terminology#STREAM) protocol.
+
+This feature allows clients to setup payments, for uses cases such
+[web monetization](./web-monetization), with a single API call, to get the
+payment details, and be guaranteed that the receiving account supports ILP as
+payment method.
+
+A **PayID** is an alternative identifier which, unlike Payment Pointers that
+identify accounts, identifies [account owners](./terminology#account-owner).
+
+The
+[PayID URI specification](https://github.com/xpring-eng/rfcs/blob/master/payid/dist/spec/payid-uri.txt)
+defines the format of a PayID to be `acctpart "$" host`.
+
+With respect to Open Payments the `acctpart` is the account owner identifier and
+`host` is the origin of the Open Payments server.
```

Comment:
We've seen a lot of partners host their PayID server at specific subdomains (ie payid.wallet.com), which results in PayIDs that look like `nkramer$payid.wallet.com`.  I think we want to make it clear that the `host` portion of a PayID will most likely **not** be the same as the host of the Open Payments server.

A likely scenario is that PayID and Open Payments operators will put a proxy in front of the two respective servers, and proxy requests to the correct server based on something in the request.  To reduce as much complexity in the proxy logic as possible, I think it makes sense to put Open Payments discovery at something like `/{acctPart}/.well-known/open-payments`.  This makes it abundantly clear to the proxy that that request is meant for the Open Payments server.  Once a client has done discovery on a PayID, they can do Open Payments things at specific account URLs.

For example, given a PayID of `nkramer$wallet.com`, a client can make a request to `https://wallet.com/nkramer/.well-known/open-payments`, which will be proxied to `https://openpayments.wallet.com/nkramer/.well-known/open-payments`.  This will return account URLs such as `https://openpayments.wallet.com/nkramerXRPaccount`.

Reply:
> I think we want to make it clear that the host portion of a PayID will most likely not be the same as the host of the Open Payments server.

Why? What are the functions of a PayID server that they need to be hosted independently of an Open Payments server?

Reply:
Imagine a wallet is running Xpring's PayID server implemented in JS, and running our OPS which is implemented in Java.  I'm not sure how you would host two different runtimes on the same host without placing a proxy in front of both of them. 

My point is that in order for your PayID to allow OP discovery, there would need to be a proxy sitting in front of both the PayID server and the OP server.  Without a defined OP Discovery namespace, the proxy's routing logic becomes really complicated.

Reply:
>  the proxy's routing logic becomes really complicated.

I will second this feedback. I think we're _wildly_ underestimating how difficult it will be for any mature organization to "just deploy a proxy" for this kind of stuff.

Two trivial examples: The last company I worked for suffered outages due to proxy-config complexity and had very strict rules about **not** putting API business logic into the edges. They tried _very hard_ to keep this logic in the API applications themselves because this was the best place for that logic to live (especially from a QA perspective). Getting something like this into their proxy/LB config would have taken months, and lots of $$ for all the teams to collaborate around the changes because proxy changes tend to be organizationally-cross-cutting (i.e., everyone has a thing in the proxy). Not to mention that the proxy changes were always high-risk because there wasn't a single QA system in-place for this layer of the infrastructure, because it was cross-cutting. In short, the rule was "don't put stuff into the edge layer"

Another example that comes to mind is "any large company". E.g., imagine if you had to deploy this to https://ibm.com. How long do you think their "just tweak the proxy" process is?

Reply:
I still don't understand why a wallet would run a PayID server and an Open Payments server? As I said above:

> What are the functions of a PayID server that they need to be hosted independently of an Open Payments server?

If I look at this through the lens of a wallet that already implements Open Payments then I see PayID as a new feature I am adding to my wallet.

So I would issue my users with a PayID and tell them they can use this to enable other wallets to pay them.

So then my users use their PayID somewhere and the other system (the Open Payments client) follows the discovery flow I've defined here to discover all the user's accounts and they use Open Payments to complete a transaction with them.